account-decoder/src/parse_token.rs

```diff
@@ -86,13 +98,31 @@ pub enum TokenAccountType {
 pub struct UiTokenAccount {
     pub mint: String,
     pub owner: String,
-    pub amount: u64,
+    pub token_amount: UiTokenAmount,
     pub delegate: Option<String>,
     pub is_initialized: bool,
     pub is_native: bool,
     pub delegated_amount: u64,
 }
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct UiTokenAmount {
+    pub ui_amount: f64,
+    pub decimals: u8,
+    pub amount: UiAmount,
```

Comment:
Seeing `ui_amount` as a field name and then `UiAmount` as a type kinda broke my brain.   

`UiAmount` -> `StringAmount`?  `UnconstrainedAmount`?

Reply:
Fair point! That was an Rpc-> Ui replace that definitely needed circling back on. I like `StringAmount`

Reply:
I also think that type should probably live somewhere else. I needed it lower than rpc (core), so that's why it ended up here, but once we use it for other balances, perhaps a better home will present itself.